File: TypeScript/Slice/remove-everything-from-slice.ts
function main(): void {
    // Create a new slice.
    const randomSlice: string[] = ["a", "b", "c", "d", "e"]
    // Remove everything from slice.
    console.log(removeEverythingFromSlice(randomSlice))
}

main()

// Remove everything from slice and return the slice.
function removeEverythingFromSlice(originalSlice: string[]): string[] {
    while (originalSlice.length > 0) {
        originalSlice.pop()
    }
    return originalSlice
}

File: Dart/Slice/remove_everything_from_slice.dart
import "dart:core";

void main() {
  // Create a new list of strings.
  final List<String> randomSlice = <String>["Bob", "Alice", "Joe"];
  // Remove everything from the slice.
  print(removeEverythingFromSlice(randomSlice));
}

// Remove everything from a given slice.
List<dynamic> removeEverythingFromSlice(final List<dynamic> slice) {
  slice.clear();
  return slice;
}
